src/main/java/com/squareup/catalog/demo/example/clone/CatalogObjectCloneUtil.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.catalog.demo.example.clone;
+
+import com.squareup.connect.models.CatalogObject;
+import com.squareup.connect.models.Money;
+import java.util.Collections;
+import java.util.UUID;
+
+/**
+ * Defines utility methods used when cloning a CatalogObject from one merchant account to another.
+ *
+ * @param <T> The catalog object data type
+ */
+abstract class CatalogObjectCloneUtil<T> {
+
+  final CatalogObject.TypeEnum type;
+
+  CatalogObjectCloneUtil(CatalogObject.TypeEnum type) {
+    this.type = type;
+  }
+
+  /**
+   * Returns the type specific data from a catalog object.
+   */
+  abstract T getCatalogData(CatalogObject catalogObject);
+
+  /**
+   * Encodes a subset of fields from the {@link CatalogObject} into a string, allowing
+   * CatalogObjects to be compared between merchant accounts for likeness. If the encoded string of
+   * a {@link CatalogObject} in the source account matches the encoded string of a {@link
+   * CatalogObject} in the target account, then the object is not cloned.
+   */
+  final String encodeCatalogObject(CatalogObject catalogObject) {
+    T catalogData = getCatalogData(catalogObject);
+    return encodeCatalogData(catalogData);
+  }
+
+  /**
+   * Encodes a subset of fields from the {@link CatalogObject} into a string, allowing
+   * CatalogObjects to be compared between merchant accounts for likeness. If the encoded string of
+   * a {@link CatalogObject} in the source account matches the encoded string of a {@link
+   * CatalogObject} in the target account, then the object is not cloned.
+   *
+   * @param catalogData the data from the {@link CatalogObject}.
+   */
+  abstract String encodeCatalogData(T catalogData);
+
+  /**
+   * Encodes a {@link Money} object to a string.
+   *
+   * @param money the {@link Money} object to encode
+   * @return the amount as a string, or null if the {@link Money} object is null
+   */
+  String amountOrNull(Money money) {
+    return (money == null) ? "null" : money.getAmount().toString();
+  }
+
+  /**
+   * Removes meta data from the {@link CatalogObject} that only applies to the source account, such
+   * as location IDs and version.
+   *
+   * @param catalogObject the {@link CatalogObject} to modify
+   */
+  void removeSourceAccountMetaData(CatalogObject catalogObject) {
```

Comment:
Enabled/disabled is handled (that's the location IDs).  I'll have to handle price overrides when I add support for items, but for now all this example supports is taxes, modifiers, and discounts, and they don't have price overrides.